src/StateMachine.cpp

```diff
@@ -152,6 +163,11 @@ StateMachine::Update StateMachine::processSDOReceive(canbus::Message const& msg)
         if (msg.time.isNull()) {
             throw ProtocolError("received CAN message with zero timestamp");
         }
+        if (cmd.size == 0 && useUnknownSizes) {
```

Comment:
I wouldn't have guarded this with the flag. It's part of the protocol, so it could be transparent IMO.